samples/test/utils/kfp/samples/test/utils.py

```diff
@@ -253,19 +244,28 @@ def main(
                     logger.warning(
                         f'KFP_OUTPUT_DIRECTORY env var is left for backward compatibility, please use KFP_PIPELINE_ROOT instead.'
                     )
+        logger.info(f'KFP_PIPELINE_ROOT={pipeline_root}')
         if metadata_service_host is None:
             metadata_service_host = os.getenv('METADATA_GRPC_SERVICE_HOST',
                                               'metadata-grpc-service')
+        logger.info(f'METADATA_GRPC_SERVICE_HOST={metadata_service_host}')
         if launcher_v2_image is None:
             launcher_v2_image = os.getenv('KFP_LAUNCHER_V2_IMAGE')
             if not launcher_v2_image:
                 raise Exception("launcher_v2_image is empty")
+        logger.info(f'KFP_LAUNCHER_V2_IMAGE={launcher_v2_image}')
         if driver_image is None:
             driver_image = os.getenv('KFP_DRIVER_IMAGE')
             if not driver_image:
                 raise Exception("driver_image is empty")
-
-        client = kfp.Client(host=host)
+        logger.info(f'KFP_DRIVER_IMAGE={driver_image}')
+        client = kfp.Client()
+        # TODO(Bobgy): avoid using private fields when getting loaded config
+        kfp_endpoint = client._existing_config.host
+        kfp_ui_endpoint = client._uihost
+        logger.info(f'KF_PIPELINES_ENDPOINT={kfp_endpoint}')
+        if kfp_ui_endpoint != kfp_endpoint:
```

Comment:
I'm not sure I understand, what's the difference? When the condition is true, kfp_ui_endpoint is equal to kfp_endpoint.

Reply:
Since what we do if the `if` is evaluated to true is just logging, then I think it is simpler to just log the kfp_ui_endpoint right away. Because people who are viewing the log doesn't necessarily know about the logic that `kfp_ui_endpoint` not showing up means it is equal to `kfp_endpoint`, or is the program shut down which caused log missing.

But it is NIT, so feel free to keep it as is.